### ISSUE TITLE
Crystal cave is too dark

### ISSUE THREAD

**Issue by Zwatotem** [SuperTux/supertux#1317]
**SuperTux version:** 0.6.1, possibly earlier
**System information:** GNU/Linux Parabola

##### Expected behavior
Dark part of Crystal cave level is dark, but visible to naked eye, so one can activate bonus block in the range of sight and light up the level.
##### Actual behavior
On some screens dark area is unplayably dark, or even completely black.
##### Steps to reproduce actual behavior
Play Crystal cave level, go behind the door opened by 2 switches.

**Comment by Zwatotem** [comment]
Two ways of solving this issue:
1. Make a slightly glowing circle around the penguin, not brighter, than the rest of the level, but bright enough, so everything can be seen easily. This circle should of course be a lot less convenient than other lights (literally inconvenient), that players would prefer to use alternatives for them by himself.
2. Make the whole section of the level brighter.

I'm in favor of first option, but maybe there are some other suggestions.

**Comment by HybridDog** [comment]
I am against the first option because somebody may want to create a level which is completely dark and the player has to play blindly.

The perceived screen brightness depends on the brightness of the surrounding area (in my experience), so dark areas are a lot brighter at night.
Furthermore, humans perceive colours differently at night: https://en.wikipedia.org/wiki/Mesopic_vision

Is this the right crystal level?
![cav](https://user-images.githubusercontent.com/3192173/72173343-9cceb880-33d7-11ea-83cd-b82f92eb0503.jpg)

**Comment by Zwatotem** [comment]
Yes, this is the level. I mean the part to the right behind double door.

**Comment by HybridDog** [comment]
> Yes, this is the level. I mean the part to the right behind double door.

![fot_2020-01-18-17:39:05_479007068](https://user-images.githubusercontent.com/3192173/72667226-d62ba780-3a19-11ea-9f07-5e7a9eba9a66.jpg)

This is too dark for me, too. You could add mushrooms which glow when Tux walks on them; it would have an effect similar to making Tux glow. The level designer decides where to place them, so it is still possible to keep some dark areas. In comparison to torches, the mushrooms would only glow when Tux touches them.

**Comment by Alzter** [comment]
I don't like pitch black levels, difficulty should stem from there being a fair challenge, not because you can't see any of the level, and having the player unable to see isn't exactly what I would consider fun, and I end up just turning up my PC gamma level to compensate anyway. I say we make it more visible.

**Comment by slackstone** [comment]
If you give me a dark level I'll just open it in the editor and tweak it so I can see. Overly dark levels are no fun.  .... but,  if you fall through a trap in the floor of Nolok's laboratory, that darkness is on you.

This seems like a silly issue. I'm not a fan of complete darkness, unless it's a small area or there is a really compelling reason for the gameplay. An entire level set to black, yuck.

I don't know the exact level in question "Crystal cave level " is too vague.

Which level, filename please?

**Comment by slackstone** [comment]
Regarding Options:

1) Make a slightly glowing circle around the penguin, not brighter, than the rest of the level, but bright enough, so everything can be seen easily. This circle should of course be a lot less convenient than other lights (literally inconvenient), that players would prefer to use alternatives for them by himself.

2) Make the whole section of the level brighter.

Zwatotem - You have access now so you can fix if you'd like. Option 2 seems faster, it's what I'd do for a quick fix. Penguins see pretty well in the dark.

**Comment by Zwatotem** [comment]
@slackstone as a quick fix: yes, but I like @HybridDog's solution better, only maybe not mushrooms (they can be appropriate for forest caves) but layer of magic shining crystals accumulated on blocks.

**Comment by HybridDog** [comment]
> […] layer of magic shining crystals accumulated on blocks.

I'm fine with this, too. You don't need magic for this: https://en.wikipedia.org/wiki/Triboluminescence
When the light is due to the rubbing of Tux' feet on the crystal, then you could even show a very bright light when Tux is skidding, which would be a neat feature in my opinion.

**Comment by busybody42** [comment]
You may want to make the crystal cave brighter, like maybe 3.5 of each color, pretty much...

**Comment by HybridDog** [comment]
Here's a rublight prototype: https://github.com/HybridDog/supertux/tree/m_rublight
I've called it rublight instead of triboluminescent block because I don't know how triboluminescence works physically and when translated, that greek and latin word means rublight.
This block glows when Tux walks on it, and in certain other situations.

It uses the spotlight light texture and another unfinished texture because it's unfinished:
![fot_2020-05-22-20:26:00_664082669](https://user-images.githubusercontent.com/3192173/82698549-3ef68b00-9c6b-11ea-8846-91072e5d3acd.jpg)
I have never added a new Object to SuperTux before, so I don't know if I'm doing it right.

Now I know another solution to fix this darkness problem: Simply place a lantern before the dark area.

**Comment by HybridDog** [comment]
It glows when Tux or an enemy walks on it:
![fot_2020-06-03-19:30:20_761859312](https://user-images.githubusercontent.com/3192173/83668890-00e55980-a5d1-11ea-873d-0c71d3d1693b.jpg)
![fot_2020-06-03-19:30:25_076606317](https://user-images.githubusercontent.com/3192173/83668894-017df000-a5d1-11ea-8cd3-e9d21c822a2c.jpg)
When Tux skids, it grows very bright:
![fot_2020-06-03-19:30:31_669591309](https://user-images.githubusercontent.com/3192173/83668896-02168680-a5d1-11ea-86c3-4eb4084dbacf.jpg)

**Comment by Zwatotem** [comment]
@Rusty-Box Now, that #1416 is merged and there are new crystal tiles **and** you are in the process of redesigning this level (I assume there will be at least one more version, to break the repeatable pattern in inner tiles) I summon you to review this problem once again. Currently most of the dark part has wood floor, so I think in order for RubLight to work here from logical standpoint we'll need a graphic of wood overgrown by crystals.

**Comment by Zwatotem** [comment]
Can we close this now?
